form/create_custom_field_type.rst

```diff
@@ -66,38 +71,124 @@ check out the `ChoiceType`_ class.
     extend from ``AbstractType`` is only a convenience way of implementing the
     required ``FormTypeInterface``.
 
-There are three methods that are particularly important:
+Now you can add this form type when :doc:`creating Symfony forms </forms>`::
+
+    // src/Form/Type/OrderType.php
+    namespace App\Form\Type;
+
+    use App\Form\Type\ShippingType;
+    use Symfony\Component\Form\AbstractType;
+    use Symfony\Component\Form\FormBuilderInterface;
+
+    class OrderType extends AbstractType
+    {
+        public function buildForm(FormBuilderInterface $builder, array $options)
+        {
+            $builder
+                // ...
+                ->add('shipping', ShippingType::class)
+            ;
+        }
+
+        // ...
+    }
+
+That's all. The ``shipping`` form field will be rendered correctly in any
+template because it reuses the templating logic defined by its parent type
+``ChoiceType``. If you prefer, you can also define a template for your custom
+types, as explained later in this article.
+
+Creating Form Types Created From Scratch
+----------------------------------------
+
+Some form types are so specific to your projects that they cannot be based on
+any :doc:`existing form types </reference/form/types>` because they are too
+different. Consider an application that wants to reuse in different forms the
+following set of fields as the "postal address":
+
+.. TODO: illustration of the form field
+
+As explained above, form types are PHP classes that implement
+:class:`Symfony\\Component\\Form\\FormTypeInterface`, although it's more
+convenient to extend instead from :class:`Symfony\\Component\\Form\\AbstractType`::
+
+    // src/Form/Type/PostalAddressType.php
+    namespace App\Form\Type;
+
+    use Symfony\Component\Form\AbstractType;
+    use Symfony\Component\Form\Extension\Core\Type\FormType;
+    use Symfony\Component\OptionsResolver\OptionsResolver;
+
+    class PostalAddressType extends AbstractType
+    {
+        // ...
+
+        public function getParent()
+        {
+            return FormType::class;
```

Comment:
This method is already implemented into `AbstractType` returning also `FormType::class` by default. Maybe we should only mention that but without including it here.

Reply:
You are right! Fixed. Thanks.